src/lib/salad/hll_empirical.h

```diff
@@ -0,0 +1,43 @@
+/*
+ * SPDX-License-Identifier: BSD-2-Clause
+ *
+ * Copyright 2010-2022, Tarantool AUTHORS, please see AUTHORS file.
+ */
+#pragma once
+
+#include <stdint.h>
+
+#if defined(__cplusplus)
+extern "C" {
+#endif /* defined(__cplusplus) */
+
+/**
+ * HyperLogLog needs a bias correction for small cardianlities.
+ * The bias corrections was found empirically for precsion range
+ * defined by following constants.
+ */
+enum HLL_EMPIRICAL_LIMITS {
+	/** Minimum precision value for the HyperLogLog algorithm. */
+	HLL_MIN_PRECISION = 6,
+	/** Maximum precision value for the HyperLogLog algorithm. */
+	HLL_MAX_PRECISION = 18,
+	/** Number of available precisions. */
+	HLL_N_PRECISIONS = HLL_MAX_PRECISION - HLL_MIN_PRECISION + 1,
+};
+
+/**
+ * Return the bias correction for the raw_estimation.
+ */
+double
+hll_empirical_bias_correction(uint8_t precision, double raw_estimation);
+
+/**
+ * Return the threshold below which linear counting algorithm has a smaller
+ * error than the HyperLogLog algorithm.
+ */
+uint64_t
+hll_empirical_estimation_threshold(uint8_t precision);
```

Comment:
Looks like these functions are only used in `hll.c`. Both the header and the source file are fairly small. Why not merge them into `hll.c`?

Reply:
Done.